Kudu.Core/Infrastructure/StringExtensions.cs

```diff
@@ -28,5 +29,27 @@ public static string EscapeHashCharacter(this string str)
         {
             return str.Replace("#", Uri.EscapeDataString("#"));
         }
+
+        // Returns a string you can safely use as valid Windows or Linux file name.
+        // Does not check for reserved words.
+        public static string ToSafeFileName(this string str)
+        {
+            str = str.Trim()
+                     .Replace(' ', '-');
```

Comment:
No other trace seems to contain a space so i put it in there for consistency.

<img width="495" alt="traces" src="https://cloud.githubusercontent.com/assets/6472374/15979735/963b5044-2f6e-11e6-8758-6903c8c2ea75.PNG">